plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+#include "BaseStateDriver.cpp"
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Joint.hh>
+#include <gz/sim/Link.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/Sensor.hh>
+#include <gz/sim/System.hh>
+#include <gz/sim/Util.hh>
+#include <gz/sim/components/Sensor.hh>
+#include <gz/transport/Node.hh>
+#include <iostream>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/PolyDriverList.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+namespace gzyarp
+{
+
+class BaseState : public System, public ISystemConfigure, public ISystemPostUpdate
+{
+public:
+    BaseState()
+        : m_deviceRegistered(false)
+    {
+    }
+
+    virtual ~BaseState()
+    {
+        if (m_deviceRegistered)
+        {
+            Handler::getHandler()->removeDevice(m_deviceScopedName);
+            m_deviceRegistered = false;
+        }
+
+        if (m_baseStateDriver.isValid())
+            m_baseStateDriver.close();
+        BaseStateDataSingleton::getBaseStateDataHandler()->removeBaseLink(m_baseLinkScopedName);
+        yarp::os::Network::fini();
+    }
+
+    virtual void Configure(const Entity& _entity,
+                           const std::shared_ptr<const sdf::Element>& _sdf,
+                           EntityComponentManager& _ecm,
+                           EventManager& /*_eventMgr*/) override
+    {
+        yarp::os::Network::init();
```

Comment:
I know that this is everywhere in the plugins, but like it does not make sense to manually lock and unlock mutexes, similarly it does not make sense to manually call `yarp::os::Network::init()` and `        yarp::os::Network::fini();`, you can just add an attribute `yarp::os::Network m_yarpNetwork;` to the class, the `init` and `fini` will be called in its constructor and destructor (this is typically called RAII).

Reply:
Solved in bc9dde81de292f5c1656733de1339015b8c3dddb